processor/core/src/main/java/com/redhat/service/bridge/processor/actions/AbstractActionBeanFactory.java

```diff
@@ -0,0 +1,31 @@
+package com.redhat.service.bridge.processor.actions;
+
+import java.util.Optional;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+import com.redhat.service.bridge.infra.exceptions.definitions.user.ActionProviderException;
+
+public abstract class AbstractActionBeanFactory<T extends ActionBean> {
+
+    @Inject
+    Instance<T> instances;
+
+    private final Class<T> instanceClass;
+
+    protected AbstractActionBeanFactory(Class<T> instanceClass) {
+        this.instanceClass = instanceClass;
+    }
+
+    public T get(String actionType) {
+        return getOptional(actionType)
+                .orElseThrow(() -> new ActionProviderException(String.format("No %s found for action type '%s'", instanceClass.getSimpleName(), actionType)));
+    }
+
+    public Optional<T> getOptional(String actionType) {
```

Comment:
@kostola Personally I don't think we need the `Optional` method on this class. I'd say we support `get` only as to me it makes it clearer as to expectations on what should be provided each time we add additional Actions to the platform. For example: we shouldn't have the semantics in the Shard where I can optionally as for an `ActionInvoker`, it has to exist.

Reply:
Resolving this because this part has been refactored